### PR TITLE
Added support for Azure Blob storage upload

### PR DESCRIPTION
**Pull request by gboyvalenkov-bosch**
[#6] Support for Azure BLOB storage upload

- Files can now be uploaded to Azure cloud, using temporary credentials

Signed-off-by: Georgi Boyvalenkov <Georgi.Boyvalenkov@bosch.io>

### CODE REVIEW

File: uploaders/azure.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+)
+
+// AzureUploader handles upload to Azure Blob storage
+type AzureUploader struct {
+	endpoint  string
+	sas       string
+	container string
+}
+
+// NewAzureUploader construct new AWSUploader from the provided 'start' operation options
```
Comment (ttttodorov) [inline uploaders/azure.go:39]: typo

**Review by e-grigorov — CHANGES_REQUESTED** [review]
This PR doesn't follow the contributing guideline:
https://github.com/eclipse-kanto/file-upload/blob/main/CONTRIBUTING.md#making-your-changes

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: uploaders/azure.go (hunk 1 above)
Comment (e-grigorov) [reply uploaders/azure.go]: @ttttodorov - could you please resolve the conversation if the typo is fixed?

File: uploaders/azure.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		}
+		blobHTTPHeaders.BlobContentMD5 = []byte(md5)
+	}
+	options := azblob.HighLevelUploadToBlockBlobOption{
+		HTTPHeaders: blobHTTPHeaders,
+	}
+
+	response, err := blockBlobClient.UploadFileToBlockBlob(context.Background(), file, options) // perform upload
+	if err == nil {
+		if response.StatusCode/100 != 2 {
```
[the remark this answers is not in the document]
Comment (gboyvalenkov-bosch) [reply uploaders/azure.go:80]: Not sure how to restrict the codes, maybe only 200 should be sufficient. But according to specification:
10.2 Successful 2xx
This class of status code indicates that the client's request was successfully received, understood, and accepted.

File: go.mod [hunk 3]
```diff
@@ -16,6 +17,15 @@ require (
 )
 
 require (
+	github.com/pkg/browser v0.0.0-20180916011732-0a3d74bf9ce4 // indirect
```
[the remark this answers is not in the document]
Comment (gboyvalenkov-bosch) [reply go.mod:20]: These are test dependencies

File: uploaders/azure.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	}
+	options := azblob.HighLevelUploadToBlockBlobOption{
+		HTTPHeaders: blobHTTPHeaders,
+	}
+
+	response, err := blockBlobClient.UploadFileToBlockBlob(context.Background(), file, options) // perform upload
+	if err == nil {
+		logger.Debugf("azure blob upload response status code - %v", response.StatusCode)
+		if response.StatusCode != 201 {
+			return fmt.Errorf("unsuccessful response status code - %v", response.StatusCode)
```
Comment (e-grigorov) [inline uploaders/azure.go:83]: with StatusCode != 201, the error message should be: unsuccessful upload, response status code...

**Review by e-grigorov — CHANGES_REQUESTED** [review]

**Review by e-grigorov — APPROVED** [review]

**Pull request merged**